src/Features/Core/Portable/GenerateConstructorFromMembers/AbstractGenerateConstructorFromMembersCodeRefactoringProvider.cs

```diff
@@ -69,12 +78,20 @@ public override Task ComputeRefactoringsAsync(CodeRefactoringContext context)
             string? serializedIntentData,
             CancellationToken cancellationToken)
         {
+            Accessibility? desiredAccessibility = null;
+            if (serializedIntentData != null)
+            {
+                var intentData = JsonSerializer.Deserialize<GenerateConstructorIntentData>(serializedIntentData, _jsonSerializerOptions);
```

Comment:
It feels like a leaky abstraction to have this know about json deserialization etc. Is it overkill to have an IntentDeserializer service that hides the json bits? Or can this be moved up to a higher level (I admit I haven't looked to see where `serializedIntentData` comes from)

Reply:
I think this is better - lmk!

Reply:
LGTM, nice one